test/errors.js

```diff
@@ -32,8 +31,14 @@ function generateDisconnectionCase(readyKillIdFun, cleanupKillIdFun) {
 
     function killThread(argFun) {
       var threadId = argFun(zongji);
-      test.ok(!isNaN(threadId));
-      conn.db.query('KILL ' + threadId);
+      if (typeof threadId === 'object') {
+        threadId.end(function (err) {
+          test.done();
```

Comment:
Ok, now I call test.done() before the `ctrlPool` was killed and also I wrote the new test which checks that the `ctrlPool` is pool. See my last commit. Thanks you for feedback.